website/src/components/starlight-overrides/Header.astro

```diff
@@ -8,47 +8,78 @@ import SocialIcons from "@astrojs/starlight/components/SocialIcons.astro";
 import ThemeSelect from "@astrojs/starlight/components/ThemeSelect.astro";
 import MobileMenuToggle from "@astrojs/starlight/components/MobileMenuToggle.astro";
 import { LinkButton } from "@astrojs/starlight/components";
-import NavMenuItem from "../NavMenuItem.astro";
+import NavMenuItem, {
+  type Props as NavMenuItemProps,
+} from "../NavMenuItem.astro";
+
+export interface Props {
+  ctaButton?: {
+    url: string;
+    text: string;
+  };
+  navMenuItems?: NavMenuItemProps[];
+}
```

Comment:
Most of these changes are just rearranging the order and adding them to a `Props` interface (standard Astro pattern for managing page properties) for typesafety.